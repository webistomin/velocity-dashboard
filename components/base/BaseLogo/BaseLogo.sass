+b('base-logo')
  display: inline-block

  &:hover .base-logo__title
    color: var(--color-blue)

  +e('link')
    display: flex

  +e('title')
    font: var(--text-logo)
    transition: color 0.2s ease-out

  +e('icon')
    margin-right: rem(7px)
