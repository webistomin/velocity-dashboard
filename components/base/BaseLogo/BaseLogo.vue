<template lang="pug" functional>
  .base-logo
    nuxt-link(to='/').base-logo__link.link
      svg(xmlns='http://www.w3.org/2000/svg' width='21.136' height='23.774').base-logo__icon
        path(d='M14 0v9.333a14 14 0 01-14 14V14A14 14 0 0114 0z' fill='#2cc2a5')
        path(d='M21.136 5.788l-3.192 8.77A14 14 0 010 22.926l3.192-8.77a14 14 0 0117.944-8.368z' fill='#2e5bff')
      h1.base-logo__title.title velocity
</template>

<script>
  import Vue from 'vue';

  export default Vue.extend({
    name: 'BaseLogo',
  });
</script>

<style lang="sass">
  @import 'BaseLogo'
</style>
