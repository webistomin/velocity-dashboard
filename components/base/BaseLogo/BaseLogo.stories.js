import { BaseLogo } from './BaseLogo';

export default {
  title: 'BaseLogo',
  parameters: {
    options: { showPanel: false },
  },
};

export const Default = () => ({
  render(h) {
    return h(BaseLogo);
  },
});
