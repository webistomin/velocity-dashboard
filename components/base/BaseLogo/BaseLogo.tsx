import './BaseLogo.sass';

export const BaseLogo = () => {
  return (
    <div class="base-logo">
      <nuxt-link to="/" class="base-logo__link link">
        <svg xmlns="http://www.w3.org/2000/svg" width="21.136" height="23.774" class="base-logo__icon">
          <path d="M14 0v9.333a14 14 0 01-14 14V14A14 14 0 0114 0z" fill="#2cc2a5" />
          <path d="M21.136 5.788l-3.192 8.77A14 14 0 010 22.926l3.192-8.77a14 14 0 0117.944-8.368z" fill="#2e5bff" />
        </svg>
        <h1 class="base-logo__title title">velocity</h1>
      </nuxt-link>
    </div>
  );
};
