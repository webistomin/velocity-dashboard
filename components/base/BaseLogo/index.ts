import BaseLogo from './BaseLogo.vue';

export default BaseLogo;
