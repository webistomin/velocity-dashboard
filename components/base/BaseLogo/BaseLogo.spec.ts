import { mount } from '@vue/test-utils';
import { BaseLogo } from './BaseLogo';

describe('BaseLogo', () => {
  const factory = (propsData = {}, options = {}) => {
    return mount(BaseLogo, {
      propsData: {
        ...propsData,
      },
      stubs: ['nuxt-link'],
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });
});
