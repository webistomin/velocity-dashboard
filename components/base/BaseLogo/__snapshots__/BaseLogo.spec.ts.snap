// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseLogo Matches snapshot 1`] = `
"<div class=\\"base-logo  \\">
  <nuxt-link-stub to=\\"/\\" class=\\"base-logo__link link\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"21.136\\" height=\\"23.774\\" class=\\"base-logo__icon\\">
      <path d=\\"M14 0v9.333a14 14 0 01-14 14V14A14 14 0 0114 0z\\" fill=\\"rgb(var(--color-darkturquoise))\\"></path>
      <path d=\\"M21.136 5.788l-3.192 8.77A14 14 0 010 22.926l3.192-8.77a14 14 0 0117.944-8.368z\\" fill=\\"rgb(var(--color-primary))\\"></path>
    </svg>
    <h1 class=\\"base-logo__title title\\">velocity</h1>
  </nuxt-link-stub>
</div>"
`;
